src/lib/menu/menu-directive.ts

```diff
@@ -81,6 +100,12 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
     }
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['xPosition'] || changes['yPosition']) {
+      this.setPositionClasses();
```

Comment:
The classes need to be updated whenever the xPosition or the yPosition changes.  For that reason, it makes more sense to me to call this method in the xPosition and yPosition setters rather than creating a hook just to listen for changes on those properties.